### PR TITLE
SGA-5362 Add data source for DAC

### PR DESCRIPTION
**Pull request by loren-tsahalon**

### CODE REVIEW

File: docs/resources/datastore.md [hunk 1]
```diff
@@ -151,13 +151,17 @@ Required:
 ## Example Usage
 
 ```terraform
-locals {
-  dataaccess_controller_id = "<assigned dataaccess_controller_id>"
+data "satori_data_access_controller" "dac0" {
```
Comment (otlg) [inline docs/resources/datastore.md:154]: let's keep both examples

Comment (otlg) [reply docs/resources/datastore.md:154]: for example: datastore0 will use id and datastoreWithIgnorePasswordUpdate will use the new object.

If you have id there is no need to request the dac object

File: satori/api/client.go [hunk 2]
```diff
@@ -161,6 +161,34 @@ func (c *Client) getJsonForAccount(path string, search *string, output interface
...
+		return err
+	}
+
+	q := req.URL.Query()
+	for k, v := range *params {
+		q.Set(k, v)
+	}
+	q.Add("accountId", c.AccountId)
+	q.Add("page", "0")
+	q.Add("pageSize", "200")
```
Comment (otlg) [inline satori/api/client.go:176]: pageSize should be much smaller :)

File: satori/datasources/datasource_data_access_controller.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "DAC's region.",
+			},
+			"cloud_provider": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "DAC's cloud provider.",
+			},
+			"unique_name": {
```
Comment (otlg) [inline satori/datasources/datasource_data_access_controller.go:30]: After second thought, let's remove it at all.

You can query the DAC with:
1. ID (for private)
2. type region provider (for public only)

Make sense?

File: satori/datasources/datasource_data_access_controller.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "DAC's unique name. Provide when the type is PRIVATE or PRIVATE_MANAGED.",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
```
Comment (otlg) [inline satori/datasources/datasource_data_access_controller.go:37]: I mean let's remove computed, as it read-only and allow options like above

File: satori/datasources/datasource_data_access_controller.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+				Description: "DAC's ID.",
+			},
```
Comment (otlg) [inline satori/datasources/datasource_data_access_controller.go:39]: Also, let's add a computed IP list

Comment (otlg) [reply satori/datasources/datasource_data_access_controller.go:39]: Any other property we want to share?

File: satori/datasources/datasource_data_access_controller.go (hunk 3 above)
Comment (otlg) [reply satori/datasources/datasource_data_access_controller.go:30]: We can also make a validation for input

File: satori/datasources/datasource_data_access_controller.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+		return diag.Errorf("DAC with type of '%s' must include its unique name. region: '%s', cloud provider: '%s'", dacType, region, cloudProvider)
+	}
+
+	dacs, err := c.QueryDataAccessControllers(&dacType, &region, &cloudProvider, &uniqueName)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	found := false
+	if len(dacs) > 0 {
```
Comment (otlg) [inline satori/datasources/datasource_data_access_controller.go:64]: I think that len should be 1 only, if the len > 1, return the error that the query is not specific enough.

You can get different results 2 runs

File: satori/api/client.go (hunk 2 above)
Comment (loren-tsahalon) [reply satori/api/client.go:176]: It's a generic function and not only for this usage. What do you think should be the size? 
I saw that we set this size also in `getJsonForAccount`

**Review by otlg — APPROVED** [review]

**Pull request merged**